kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -358,13 +367,22 @@ private static void registerFlushListenerForStoreBuilders(
    * each commit, similar to #flushCache
    */
   public void flushAndAwaitPendingEvents() {
+    if (processingException != null) {
+      // if there was a processing exception, processing for the key that hit the processing
+      // exception is blocked, so we don't want to go into the loop below that tries to drain
+      // queues, as the queue for that key will never drain. Instead, just throw here.
```

Comment:
yep, fair enough. I agree that we want to throw immediately for all those reasons. I was trying to point out that you don't want to move this check because you'll probably dead hang if you do, which seems really bad - but honestly doing anything other than dropping dead when you're in an undefined state is potentially just as bad and can lead to corruption. I'll clarify.